src/error.rs

```diff
@@ -1,6 +1,8 @@
-use std::fmt;
-use std::io;
+use std::{fmt, io};
 
+pub type Result<T, E = PipelineError> = std::result::Result<T, E>;
```

Comment:
This makes our function signatures a little cleaner, and is a pretty common thing to do.